tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -740,6 +740,25 @@ void ROOT::Experimental::Internal::RPagePersistentSink::UpdateSchema(const RNTup
                                                                      NTupleSize_t firstEntry)
 {
    const auto &descriptor = fDescriptorBuilder.GetDescriptor();
+
+   if (descriptor.GetNLogicalColumns() > descriptor.GetNPhysicalColumns()) {
+      // If we already have alias columns, add an offset to the alias columns so that the new physical columns
+      // of the changeset follow immediately the already existing physical columns
+      auto getNColumns = [](const RFieldBase &f) -> std::size_t {
+         const auto &reps = f.GetColumnRepresentatives();
+         if (reps.empty())
+            return 0;
+         return reps.size() * reps[0].size();
+      };
+      std::uint32_t nNewPhysicalColumns = 0;
+      for (auto f : changeset.fAddedFields) {
+         nNewPhysicalColumns += getNColumns(*f);
+         for (const auto &descendant : *f)
+            nNewPhysicalColumns += getNColumns(descendant);
+      }
+      fDescriptorBuilder.ShiftAliasColumns(nNewPhysicalColumns);
```

Comment:
Sorry to ask about this again, I forgot the answer: Do we already support adding new column representations to existing fields? Do we want to support this eventually?

Reply:
Yes, it is supported by the s11n and tested here: https://github.com/root-project/root/blob/master/tree/ntuple/v7/test/ntuple_serialize.cxx#L1402

But not yet used by the merger.